sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintRuleRepository.java

```diff
@@ -41,10 +44,34 @@ public PylintRuleRepository(RulesDefinitionXmlLoader xmlLoader) {
   @Override
   public void define(Context context) {
     NewRepository repository = context
-        .createRepository(REPOSITORY_KEY, Python.KEY)
-        .setName(REPOSITORY_NAME);
-    xmlLoader.load(repository, getClass().getResourceAsStream(RULES_FILE), Charsets.UTF_8.name());
-    SqaleXmlLoader.load(repository, SQALE_FILE);
+      .createRepository(REPOSITORY_KEY, Python.KEY)
+      .setName(REPOSITORY_NAME);
+    xmlLoader.load(repository, getClass().getResourceAsStream(RULES_FILE), StandardCharsets.UTF_8.name());
+    defineRemediationFunction(repository);
     repository.done();
   }
+
+  private static void defineRemediationFunction(NewRepository repository) {
+    Map<String, String> remediationCostMap = loadRemediationCostMap();
+    for (NewRule rule : repository.rules()) {
+      String gap = remediationCostMap.get(rule.key());
+      if (gap == null) {
+        throw new IllegalStateException("Missing remediation cost for rule " + rule.key());
+      } else if (!gap.equals("null")) {
+        rule.setDebtRemediationFunction(rule.debtRemediationFunctions().linear(gap));
+      }
+    }
+  }
+
+  private static Map<String, String> loadRemediationCostMap() {
+    Map<String, String> map = new HashMap<>();
+    try (Scanner scanner = new Scanner(PylintRuleRepository.class.getResourceAsStream(REMEDIATION_FILE))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of constructor "Scanner(InputStream)" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)